### ISSUE TITLE
External responses with $ref headers fail to load 

### ISSUE THREAD

**Issue by scottr-ad** [swagger-api/swagger-parser#1177]
For responses located in a externally referenced file, if the response has $ref header elements pointing to a different external file, the $ref headers fail to load correctly.

swagger.yaml
---------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: test
paths:
  /update-pets:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: 'common/common-types.yaml#/components/schemas/PetsList'
      responses:
        '200':
          $ref: 'common/responses/common-responses.yaml#/components/responses/PetsListResponse'
```

common/common-types.yaml
----------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: "Common Data Types"
paths: {}
components:
  schemas:
    PetsList:
      type: array
      items:
        $ref: '#/components/schemas/Pet'

    Pet:
      type: object
      properties:
        petType:
          type: string
        breed:
          $ref: '#/components/schemas/Breed'

    Breed:
      type: object
      properties:
        name:
          type: string
        family:
          type: string
```

common/responses/common-responses.yaml
----------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: "Common Responses"
paths: {}
components:
  responses:
    PetsListResponse:
      description: returns a list of pets
      headers:
        x-session-id:
          $ref: '../../headers/common-headers.yaml#/components/headers/x-session-id'
      content:
        application/json:
          schema:
            $ref: '../common-types.yaml#/components/schemas/PetsList'
```

headers/common-headers.yaml
----------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: "Common Headers"
paths: {}
components:
  headers:
    x-session-id:
      schema:
        type: string
```

In the example above, the parser fails with a RuntimeException message "Could not find ../../headers/common-headers.yaml on the classpath"

The remedy is to modify ExternalRefProcessor.processRefToExternalResponse to drill down into the sub-elements (eg. headers, content, etc) and load them while it still has a reference to the original file it is loaded from.

**Comment by scottr-ad** [comment]
Pull Request #1178 raised to fix this issue.

@gracekarina - could you (or one of the other regular contributors) please review this and approve or provide feedback?

**Comment by gracekarina** [comment]
thanks for the PR. This issue was closed by the merge of #1178
